src/pages/trade/api/book.ts

```diff
@@ -1,27 +1,28 @@
 import { useQuery } from '@tanstack/react-query';
 import { useRefetchOnNewBlock } from '@/shared/api/compact-block.ts';
-import { RouteBookResponse, RouteBookResponseJson } from '@/shared/api/server/book/types';
+import { RouteBookResponse } from '@/shared/api/server/book/types';
 import { deserializeRouteBookResponseJson } from '@/shared/api/server/book/serialization.ts';
+import { RouteBookApiResponse } from '@/shared/api/server/book';
+import { usePathSymbols } from '@/pages/trade/model/use-path.ts';
 
-export const useBook = (symbol1: string | undefined, symbol2: string | undefined) => {
+export const useBook = () => {
+  const { baseSymbol, quoteSymbol } = usePathSymbols();
   const query = useQuery({
-    queryKey: ['book', symbol1, symbol2],
+    queryKey: ['book', baseSymbol, quoteSymbol],
     queryFn: async (): Promise<RouteBookResponse> => {
-      if (!symbol1 || !symbol2) {
-        throw new Error('Missing symbols');
-      }
-
       const paramsObj = {
-        baseAsset: symbol1,
-        quoteAsset: symbol2,
+        baseAsset: baseSymbol,
+        quoteAsset: quoteSymbol,
       };
       const baseUrl = '/api/book';
       const urlParams = new URLSearchParams(paramsObj).toString();
       const res = await fetch(`${baseUrl}?${urlParams}`);
-      const data = (await res.json()) as RouteBookResponseJson;
-      return deserializeRouteBookResponseJson(data);
+      const jsonRes = (await res.json()) as RouteBookApiResponse;
+      if ('error' in jsonRes) {
+        throw new Error(jsonRes.error);
+      }
```

Comment:
Because nextjs can return an error field, we need to parse it and re-throw if we see it